src/async_impl/client.rs

```diff
@@ -1737,8 +1755,17 @@ impl ClientBuilder {
     /// # Optional
     ///
     /// This requires the `rustls-tls(-...)` Cargo feature enabled.
-    #[cfg(feature = "__rustls")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "rustls-tls")))]
+    #[cfg(all(
+        feature = "__rustls",
+        not(feature = "rustls-tls-platform-verifier-no-provider")
+    ))]
```

Comment:
I don't think we can do _this_, since it could that if a user calls `add_crl()`, and then adds a dependency that happens to also depend on `reqwest/rustls-tls-platform-verifier-no-provider`, then suddenly the type and methods will go away cause compilation breakage.

They'd still need to exist, just document (and perhaps warn) that they aren't used if the `.with_platform_verifier()` code is entered. Presumably, just because something else has enabled the crate feature, a user could specify on the `ClientBuilder` what to actually use.

Reply:
Sorry for the lack of movement in this PR, I’ve been quite busy with job interviews.

There are multiple challenges with this approach. Starting by the fact that we won’t be able to pin rustls-platform-verifier to a compatible version for MSRV, the current version and the one compatible with Rust 1.64 have completely different interfaces.

I’ll take another look at this to see if there is a way to make it work.

@seanmonstar have you considered keep using WebPKI as the verifier while making Rustls the default TLS stack? That might actually be a more expected behavior than switching verifiers complete since it’s what you get now when you enable the `rustls` feature.